File: Array&String/delete_columns_to_make_sorted.py
class Solution:
    def minDeletionSize(self, strs: list[str]) -> int:
        n = len(strs[0])
        idxs = set(range(len(strs[0])))
        cur = strs[0]
        for s in strs[1:]:
            stack = set()
            for idx in idxs:
                if s[idx] < cur[idx]: stack.add(idx)
            idxs -= stack
            cur = s
        return n - len(idxs)

    # shorter
    def minDeletionSize(self, strs: list[str]) -> int:
        count = 0
        for i in zip(*strs):
            if list(i) != sorted(i):
                count += 1
        return count

File: Array&String/CPP/delete_columns_to_make_sorted.cpp
#include <algorithm>
#include <numeric>
#include <string>
#include <vector>
using namespace std;

class Solution {
public:
    int minDeletionSize(vector<string>& strs) {
        int n = strs[0].size();
        vector<int> idxs(n);
        string cur = strs[0];
        iota(idxs.begin(), idxs.end(), 0);
        for (int i = 1; i < strs.size(); i++) {
            vector<int> stack = idxs;
            for (int idx : idxs) {
                if (strs[i][idx] < cur[idx]) {
                    auto it = find(stack.begin(), stack.end(), idx);
                    stack.erase(it);
                }
            }
            idxs = stack;
            cur = strs[i];
        }
        return n - idxs.size();
    }
};